pkg/datapath/linux/node_addressing.go

```diff
@@ -36,8 +36,7 @@ func listLocalAddresses(family int) ([]net.IP, error) {
 		return nil, err
 	}
 
-	var addresses []net.IP
-
+	addresses := make([]net.IP, 0, len(addrs)-len(ipsToExclude))
```

Comment:
please add a comment here, I had to read the code to understand why this subtraction was happening.

Reply:
Thinking more about this case, it might be that `addrs` might not contain any address to exclude while `ipsToExclude` is non-empty. In this case we would preallocate to few entries. So I think it might be better to just drop this change, instead of trying to get it right in all cases.